# Why: tester_main.cc relied on runtime and tonic headers that came
# transitively before the Dart runtime moved out of the shell.
diff --git a/engine/src/flutter/shell/testing/BUILD.gn b/engine/src/flutter/shell/testing/BUILD.gn
index 7580937f9d0..9304e5326ef 100644
--- a/engine/src/flutter/shell/testing/BUILD.gn
+++ b/engine/src/flutter/shell/testing/BUILD.gn
@@ -72,6 +72,7 @@ template("tester") {
       "//flutter/flow",
       "//flutter/fml",
       "//flutter/lib/snapshot",
+      "//flutter/runtime",
       "//flutter/shell/common",
       "//flutter/shell/common:dart",
       "//flutter/shell/gpu:gpu_surface_software",
diff --git a/engine/src/flutter/shell/testing/tester_main.cc b/engine/src/flutter/shell/testing/tester_main.cc
index 568f16652ab..6a779d513b3 100644
--- a/engine/src/flutter/shell/testing/tester_main.cc
+++ b/engine/src/flutter/shell/testing/tester_main.cc
@@ -18,6 +18,8 @@
 #include "flutter/fml/paths.h"
 #include "flutter/fml/synchronization/waitable_event.h"
 #include "flutter/fml/task_runner.h"
+#include "flutter/runtime/dart_isolate.h"
+#include "flutter/runtime/isolate_configuration.h"
 #include "flutter/shell/common/platform_view.h"
 #include "flutter/shell/common/rasterizer.h"
 #include "flutter/shell/common/shell.h"
@@ -29,6 +31,8 @@
 #include "third_party/dart/runtime/include/bin/dart_io_api.h"
 #include "third_party/dart/runtime/include/dart_api.h"
 #include "third_party/skia/include/core/SkSurface.h"
+#include "third_party/tonic/converter/dart_converter.h"
+#include "third_party/tonic/logging/dart_error.h"
 
 #include "flutter/shell/testing/tester_context.h"
 #include "flutter/shell/testing/tester_context_gles_factory.h"
